### PR TITLE
Add openshift-dr-ops namespce to DR collection script

### PR DESCRIPTION
**Pull request by OdedViner**
At the development team's request, we have added the [openshift-dr-op] namespace, and we will collect data from it if it exists. https://bugzilla.redhat.com/show_bug.cgi?id=2291180

### CODE REVIEW

**Comment by black-dragon74** [comment]
@OdedViner was there a discussion related to this? Also, please provide details (logs) if you tested it.

**Comment by OdedViner** [comment]
> @OdedViner was there a discussion related to this? Also, please provide details (logs) if you tested it.

```
#!/usr/bin/env bash

# Expect base collection path as an exported variable
# If it is not defined, use PWD instead

# Get all namespaces
NAMESPACE_NAMES=("openshift" "openshift1" "openshift2" "openshift-dr-ops" "openshift3"  "openshift-operators" "openshift-4")

# Items to check for
NAMESPACES_DR=("openshift-dr-system" "openshift-operators" "openshift-dr-ops") 

NAMESPACES=()

# Iterate through DR namespaces
for NAMESPACE_DR in "${NAMESPACES_DR[@]}"; do
    # Check if the DR namespace exists in the list of all namespaces
    for NAMESPACE_NAME in "${NAMESPACE_NAMES[@]}"; do
        if [ "$NAMESPACE_NAME" == "$NAMESPACE_DR" ]; then
            # Append the DR namespace to NAMESPACES if it exists
            NAMESPACES+=("$NAMESPACE_DR")
        fi
    done
done

for NAMESPACE in "${NAMESPACES[@]}"; do
    echo $NAMESPACE
done

$ sh mg_dev.sh 
openshift-operators
openshift-dr-ops
```
I will create new image and test it on ODF cluster

**Comment by OdedViner** [comment]
> @OdedViner please check the ci failures. Also good to tag someone from DR team to review, Maybe the bug reporter

@yati1998 CI failures fixed

**Comment by raghavendra-talur** [comment]
@OdedViner Thank you for the fix! I have one minor comment for the changes made.

Also, we have a new resource type related to DR which needs to be added to `gather_namespaced_resources` file under `# collect get output of OC commands for all namespaces` and `# collect yaml output of OC commands`

recipe or recipe.ramendr.openshift.io

**Comment by raghavendra-talur** [comment]
> @yati1998 @raghavendra-talur I added `recipe.ramendr.openshift.io` to relevant lists. [I did not test it on live cluster] Do you think we need to add a condition?
> 
> ```
> if cluster_type is DR
>    add  recipe.ramendr.openshift.io to list
> ```

Yes, the recipe CRD is only available on DR clusters.

File: collection-scripts/gather_dr_resources [hunk 1]
```diff
@@ -5,43 +5,49 @@
...
+# Iterate through DR namespaces
+for NAMESPACE_DR in "${NAMESPACES_DR[@]}"; do
+    # Check if the DR namespace exists in the list of all namespaces
+    for NAMESPACE_NAME in "${NAMESPACE_NAMES[@]}"; do
+        if [[ $NAMESPACE_NAME =~ $NAMESPACE_DR ]]; then
+            # Append the DR namespace to NAMESPACES if it exists
+            NAMESPACES+=("$NAMESPACE_DR")        
+        fi
+    done
+done
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_dr_resources:25]: done

**Comment by black-dragon74** [comment]
What if the current cluster consumes the storage? Is that accounted for as well?

**Comment by OdedViner** [comment]
> @OdedViner was there a discussion related to this? Also, please provide details (logs) if you tested it.
done https://github.com/red-hat-storage/odf-must-gather/pull/196#issuecomment-2356167389

**Comment by OdedViner** [comment]
> We do not need a conditional in order to add `recipe`. It is fine to add it to the list unconditionally, it will fail silently and will be logged. This will save us from polluting the exports.
> 
> Rename variables for ease of readability:
> 
> * `NAMESPACE_NAMES`  to `ALL_NAMESPACES`
> * `NAMESPACES_DR`  to `DR_NAMESPACES`
> * `NAMESPACE_DR` to `DR_NAMESPACE`
> * `NAMESPACE_NAME` to `NAMESPACE`
> 
> Use `dbglog` helper for printing to console (it also appends to the log file). Using `echo` just prints to console.
> 
> P.S: Please test the code locally before pushing it to remote.
> 
> Regards

done

File: collection-scripts/gather_dr_resources [hunk 2]
```diff
@@ -5,19 +5,27 @@
...
-NAMESPACE_OPERATORS="openshift-operators"
-
-# Check if item1 exists in the output
-if [[ $NAMESPACES =~ $NAMESPACE_DR_SYSTEM ]]; then
-    NAMESPACE=$NAMESPACE_DR_SYSTEM
-elif [[ $NAMESPACES =~ $NAMESPACE_OPERATORS ]]; then
-    NAMESPACE=$NAMESPACE_OPERATORS
-else
-    echo "Neither $NAMESPACE_DR_SYSTEM nor $NAMESPACE_OPERATORS exists in the namespaces"
+ALL_NAMESPACES=$(oc get namespace | awk '{print $1}')
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_dr_resources]: done

File: collection-scripts/gather_namespaced_resources [hunk 3]
```diff
@@ -96,6 +98,8 @@ oc_get_all+=("drivers.csi.ceph.io")
 oc_get_all+=("cephconnections.csi.ceph.io")
 oc_get_all+=("clientprofiles.csi.ceph.io")
 oc_get_all+=("clientprofilemappings.csi.ceph.io")
+if is_dr_cluster; then
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_namespaced_resources]: done

File: collection-scripts/gather_dr_resources [hunk 4]
```diff
@@ -5,19 +5,27 @@
...
+# Iterate through DR namespaces
+for DR_NAMESPACE in "${DR_NAMESPACES[@]}"; do
+    # Check if the DR namespace exists in the list of all namespaces
+    for NAMESPACE in "${ALL_NAMESPACES[@]}"; do
+        if [[ $NAMESPACE =~ $DR_NAMESPACE ]]; then
+            # Append the DR namespace to NAMESPACES if it exists
+            NAMESPACES+=("$DR_NAMESPACE")        
+        fi
+    done
+done
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_dr_resources]: done

File: collection-scripts/gather_dr_resources [hunk 5]
```diff
@@ -5,43 +5,46 @@
...
+for NAMESPACE in "${NAMESPACES[@]}"; do
+    dbglog "collecting dump of openshift-dr-system namespace"
+    { oc adm inspect --dest-dir="${BASE_COLLECTION_PATH}" ${LOG_FILTER_ARGS:+"${LOG_FILTER_ARGS}"} ns/"${NAMESPACE}" 2>&1; } | dbglog
+
+    # Create the dir for oc_output for openshift-dr-system namespace
+    mkdir -p "${BASE_COLLECTION_PATH}/namespaces/${NAMESPACE}/oc_output/"
+
+    # Run the Collection of Resources to list
+    for dr_resource in "${dr_resources[@]}"; do
+        dbglog "collecting oc command ${dr-resource}"
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_dr_resources]: done

**Comment by OdedViner** [comment]
Bash Testing:
```
#!/usr/bin/env bash

# Get all namespaces
ALL_NAMESPACES=$(oc get namespace -o jsonpath='{.items[*].metadata.name}')

# DR namespaces
DR_NAMESPACES=("openshift-dr-system" "openshift-operators" "openshift-dr-ops" "openshift-network-operator") 

NAMESPACES=()

# Iterate through DR namespaces
for DR_NAMESPACE in "${DR_NAMESPACES[@]}"; do
    if echo "${ALL_NAMESPACES[@]}" | grep -q "$DR_NAMESPACE"; then
        NAMESPACES+=("$DR_NAMESPACE")
    fi
done

echo "${NAMESPACES[@]}"
```

```
$ sh test/mg_dev.sh 
openshift-operators openshift-network-operator
```


Build Privat image
```
podman build -t quay.io/oviner/ocs-must-gather:mg_dr_new_11 -f Dockerfile .
podman push quay.io/oviner/ocs-must-gather:mg_dr_new_11
```



Tested on non-dr cluster:
```
oc adm must-gather --image=quay.io/oviner/ocs-must-gather:mg_dr_new_11 -- /usr/bin/gather --dr

$ tree -L 3
.
├── event-filter.html
├── gather-debug.log
├── namespaces
│   └── openshift-operators
│       ├── apps
│       ├── apps.openshift.io
│       ├── autoscaling
│       ├── batch
│       ├── build.openshift.io
│       ├── core
│       ├── discovery.k8s.io
│       ├── image.openshift.io
│       ├── k8s.ovn.org
│       ├── monitoring.coreos.com
│       ├── networking.k8s.io
│       ├── oc_output
│       ├── openshift-operators.yaml
│       ├── policy
│       └── route.openshift.io
├── pod-metadata.json
└── timestamp
```

File: collection-scripts/gather_dr_resources [hunk 6]
```diff
@@ -5,43 +5,46 @@
...
-elif [[ $NAMESPACES =~ $NAMESPACE_OPERATORS ]]; then
-    NAMESPACE=$NAMESPACE_OPERATORS
-else
-    echo "Neither $NAMESPACE_DR_SYSTEM nor $NAMESPACE_OPERATORS exists in the namespaces"
-    exit 0
-fi
+NAMESPACE_NAMES=$(oc get namespace | awk '{print $1}')
+
+# DR namespaces
+NAMESPACES_DR=("openshift-dr-system" "openshift-operators" "openshift-dr-ops")
```
[the remark this answers is not in the document]
Comment (OdedViner) [reply collection-scripts/gather_dr_resources]: @raghavendra-talur ?

**Comment by black-dragon74** [comment]
Thanks @OdedViner, looks much better, please update the PR description in 1st comment as well? 

@raghavendra-talur PTAL if there is need of anything else. I will add LGTM post your review.

cc: @yati1998

**Comment by OdedViner** [comment]
> Thanks @OdedViner, looks much better, please update the PR description in 1st comment as well?
> 
> @raghavendra-talur PTAL if there is need of anything else. I will add LGTM post your review.
> 
> cc: @yati1998

@black-dragon74 added  PR description

**Review by raghavendra-talur — APPROVED** [review]

**Comment by raghavendra-talur** [comment]
@OdedViner @black-dragon74 The changes look good to me.

**Review by black-dragon74 — APPROVED** [review]
/lgtm
/approve

**Pull request merged**